source/common/runtime/runtime_impl.cc

```diff
@@ -41,10 +41,21 @@ void countDeprecatedFeatureUseInternal(const RuntimeStats& stats) {
 } // namespace
 
 bool SnapshotImpl::deprecatedFeatureEnabled(absl::string_view key, bool default_value) const {
-  // If the value is not explicitly set as a runtime boolean, trust the proto annotations passed as
-  // default_value.
-  if (!getBoolean(key, default_value)) {
-    // If either disallowed by default or configured off, the feature is not enabled.
+  // A deprecated feature is enabled if at least one of the following conditions holds:
+  // 1. A boolean runtime entry <key> doesn't exist, and default_value is true.
+  // 2. A boolean runtime entry <key> exists, with a value of "true".
+  // 3. A boolean runtime entry "envoy.features.enable_all_deprecated_features" with a value of
+  //    "true" exists, and there isn't a boolean runtime entry <key> with a value of "false".
+
+  const auto& entry = key.empty() ? values_.end() : values_.find(key);
+  if (entry == values_.end() || !entry->second.bool_value_.has_value()) {
+    // Key wasn't found, the feature is enabled if default_value is true or
+    // "envoy.features.enable_all_deprecated_features" is enabled.
+    if (!default_value && !getBoolean("envoy.features.enable_all_deprecated_features", false)) {
+      return false;
+    }
+  } else if (!entry->second.bool_value_.value()) {
```

Comment:
Could this be:
```
if (!getBoolean(key, getBoolean("envoy.features.enable_all_deprecated_features", default_value))) {
  return false;
}
```

Reply:
Yes, it's possible to change the if statement on line 54 as you suggested (although the "else if" statement on line 57 will stay as is). I'll update the code.

Reply:
Apologies, I missed the nested call to `getBoolean` in your proposal.
You are correct, and I'll update the code.